src/ansible_navigator/configuration_subsystem/navigator_configuration.py

```diff
@@ -57,7 +57,7 @@ def generate_editor_command() -> str:
     else:
         message = "EDITOR environment variable set as '{editor}'"
         initialization_messages.append(LogMessage(level=logging.DEBUG, message=message))
-        command = "%s {filename}" % os.environ.get("EDITOR")
+        command = f"{os.environ.get('EDITOR')} {{filename}}"
```

Comment:
For the conversion to be really equivalent, use the same explicit conversion specifier
```suggestion
        command = f"{os.environ.get('EDITOR')!s} {{filename}}"
```

Reply:
@webknjaz Can you show me where os.environ.get returns anything but a string?

Reply:
I think it can return `None` if there's no such key. But my comment was about a different thing — the syntax you used is not equivalent. The original string used an explicit placeholder "its a string" which for your f-string variant it's implicit. If fact, off the top of my head I can't even remember whether it's `str()` or `repr()` by default — this is why I prefer explicit over implicit.

Reply:
@webknjaz in this case, it was already checked, but we should use that instead :) (fixed)

a string used in an fstring will be rendered as a string, so I don't see the difference knowing it is a string, and the outcome of the f string is a string which I guess is implied?

Reply:
The main motivation was to point out that your conversions are not equivalent. It's best not to mix up formatting changes and potential functional changes. When we make exceptions for these rules, we tend to miss things because the focus is on the different aspect of the change. Plus from readability perspective, it's nice to say "I expect this to be a string" rather than "I hope this will get turned into a string at this point of the control flow regardless of whether something above gets refactored". When you do code changes, do you go through all the places where a variable that you're changing is going to be rendered thinking about the corner cases that may happen in the places of rendering?
Having explicit markers helps with the readability long after you've committed the change.

Reply:
Happy to make the change here. Hopefully it doesn't raise questions in the future because `os.environ.get` would be a string and the `!s` is not required.